www/docs/mp/index.php

```diff
@@ -219,37 +219,28 @@
     $title .= ' MSP, '.$MEMBER->constituency();
 }
 
-$positions = array();
-
-// Position if this is a member of the Commons
-if ($MEMBER->house(HOUSE_TYPE_COMMONS)) {
-    $position = $MEMBER->current_member(HOUSE_TYPE_COMMONS) ? 'MP' : 'Former MP';
-    if ($MEMBER->constituency()) $position .= ', ' . $MEMBER->constituency();
-    $positions[] = $position;
+$known_for = '';
+$current_offices_ignoring_committees = $MEMBER->offices('current', TRUE);
+if (count($current_offices_ignoring_committees) > 0) {
+    $known_for = $current_offices_ignoring_committees[0];
 }
 
-// Position if this is a member of NIA
-if ($MEMBER->house(HOUSE_TYPE_NI)) {
-    $position = $MEMBER->current_member(HOUSE_TYPE_NI) ? 'MLA' : 'Former MLA';
-    if ($MEMBER->constituency()) $position .= ', ' . $MEMBER->constituency();
-    $positions[] = $position;
-}
-
-// Position if this is a member of Scottish Parliament
-if ($MEMBER->house(HOUSE_TYPE_SCOTLAND)) {
-    $position = $MEMBER->current_member(HOUSE_TYPE_SCOTLAND) ? 'MSP' : 'Former MSP';
-    $position .= ', ' . $MEMBER->constituency();
-    $positions[] = $position;
-}
-
-$position = implode('; ', $positions);
-
-$current_offices = $MEMBER->offices('current', TRUE);
-$former_offices = $MEMBER->offices('previous', TRUE);
-
-// If this person has named non-committee offices, they override the default
-if (count($current_offices) > 0) {
-    $position = $current_offices[0];
+$role = '';
+foreach( array(
+    'Peer' => HOUSE_TYPE_LORDS,
+    'MP' => HOUSE_TYPE_COMMONS,
+    'MLA' => HOUSE_TYPE_NI,
+    'MSP' => HOUSE_TYPE_SCOTLAND,
+) as $noun => $house) {
+    if ($MEMBER->house($house)) {
+        if (!$MEMBER->current_member($house)) {
+            $role .= 'Former ';
+        }
+        if ($MEMBER->party_text()) {
+            $role .= $MEMBER->party_text() . ' ';
+        }
+        $role .= $noun;
```

Comment:
This produces text such as "Former DUP MP DUP MLA" or "Crossbench Peer Former Labour MP" or "Conservative MP Conservative MSP"; think it needs some punctuation as the removed code did. Also on the output, it'll stick the current constituency on the end, which might not match the last element in this list (nothing says it's putting former ones first, I mean).